File: app/src/main/java/com/example/getfood/ui/loginregister/loginfragment/LoginMvpPresenter.java
package com.example.getfood.ui.loginregister.loginfragment;

import androidx.annotation.NonNull;

import com.example.getfood.ui.base.BaseMvpPresenter;

public interface LoginMvpPresenter<V extends LoginMvpView> extends BaseMvpPresenter<V> {

    void performSignOut();

    boolean isUserEmailVerified();

    void performLogin(@NonNull String userEmail, @NonNull String password);

    void verifyUserEmail();

    void sendEmailForVerification();

    void forgotPasswordClicked(String userEmail);

    void updateToken(String token);
}

File: app/src/main/java/com/example/getfood/ui/loginregister/loginfragment/LoginMvpPresenter.kt
package com.example.getfood.ui.loginregister.loginfragment

import com.example.getfood.ui.base.BaseMvpPresenter

interface LoginMvpPresenter<V : LoginMvpView?> : BaseMvpPresenter<V> {
    fun performSignOut()
    val isUserEmailVerified: Boolean
    fun performLogin(userEmail: String, password: String)
    fun verifyUserEmail()
    fun sendEmailForVerification()
    fun forgotPasswordClicked(userEmail: String)
    fun updateToken(token: String?)
}